File: chessgame-api/src/main/java/git/dimitrikvirik/chessgameapi/model/game/ChessMessage.java
package git.dimitrikvirik.chessgameapi.model.game;

import lombok.Data;
import org.springframework.data.util.Pair;

@Data
public class ChessMessage {
    Pair<Integer, Integer> fromMove;
    Pair<Integer, Integer> toMove;
    ChessFigureColor playerColor;
    Action action;
}

File: chessgame-api/src/main/kotlin/git/dimitrikvirik/chessgameapi/model/game/ChessMessage.kt
package git.dimitrikvirik.chessgameapi.model.game

enum class ChessFigureColor {
    WHITE,
    BLACK
}

data class ChessMessage(
    var fromMove: Pair<Int, Int>,
    var toMove: Pair<Int, Int>,
    var playerColor: ChessFigureColor,
    var action: Action
)

